src/target/target.c

```diff
@@ -3046,44 +3046,41 @@ static int handle_target(void *priv)
 			is_jtag_poll_safe() && target;
 			target = target->next) {
 
-		if (!target->tap->enabled)
+		/* This function only gets called every polling_interval, so
+		 * allow some slack in the time comparison. Otherwise, if we
+		 * schedule for now+polling_interval, the next poll won't
+		 * actually happen until a polling_interval later. */
+		if (!target->tap->enabled ||
+		    power_dropout ||
+		    srst_asserted ||
+		    timeval_ms() + polling_interval / 2 < target->backoff.next_attempt)
 			continue;
```

Comment:
An attempt at improving readability:

```suggestion
                const bool time_to_poll = (timeval_ms() + polling_interval / 2) >= target->backoff.next_attempt;
		if (!target->tap->enabled ||
		    power_dropout ||
		    srst_asserted ||
		    !time_to_pool) {
			/* Do not poll this target now */
			continue;
		}
```

Reply:
Done, but slightly differently.